### ISSUE TITLE
Exception after dropping database being used

### ISSUE THREAD

**Issue by bryant1410** [dbcli/mycli#253]
The following causes a crash:

``` sql
CREATE DATABASE D;
USE D;
DROP DATABASE D;
```

**Comment by tsroten** [comment]
Thanks for reporting this @bryant1410. I'm sorry you ran into a bug! Just so we have it, I've pasted the full stack trace below. It looks like the database is successfully dropped. The error comes from the background thread that is refreshing the completion list.

```
root@localhost:foo1> drop database foo1;
Query OK, 0 rows affected
root@localhost:foo1> Exception in thread completion_refresh:
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 871, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/troten/code/mycli/mycli/completion_refresher.py", line 51, in _bg_refresh
    e.socket, e.charset, e.local_infile, e.ssl)
  File "/Users/troten/code/mycli/mycli/sqlexecute.py", line 41, in __init__g completions...
    self.connect()
  File "/Users/troten/code/mycli/mycli/sqlexecute.py", line 70, in connect
    ssl=ssl)
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/__init__.py", line 88, in Connect
    return Connection(*args, **kwargs)
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/connections.py", line 679, in __init__
    self.connect()
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/connections.py", line 891, in connect
    self._request_authentication()
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/connections.py", line 1097, in _request_authentication
    auth_packet = self._read_packet()
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/connections.py", line 966, in _read_packet
    packet.check_error()
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/connections.py", line 394, in check_error
    err.raise_mysql_exception(self._data)
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/err.py", line 120, in raise_mysql_exception
    _check_mysql_exception(errinfo)
  File "/Users/troten/.virtualenvs/mycli/lib/python3.5/site-packages/pymysql/err.py", line 115, in _check_mysql_exception
    raise InternalError(errno, errorvalue)
pymysql.err.InternalError: (1049, "Unknown database 'foo1'")
```

**Comment by amjith** [comment]
*Sigh*

I was really hoping mysql wouldn't drop a database that is currently being used, but clearly I expect too much. 😜

**Comment by j-bennet** [comment]
Funny enough, `mysql` can do it and not hiccup. I'll copy the same behavior to `mycli`.

```
root@localhost:(none)> create database t1;
Query OK, 1 row affected (0.00 sec)

root@localhost:(none)> use t1;
Database changed
root@localhost:t1> drop database t1;
Query OK, 0 rows affected (0.00 sec)

root@localhost:(none)>
```

**Comment by tsroten** [comment]
This is fixed in mycli 1.12.1, which is available via pip:
```bash
pip install -U mycli
```
